### PR TITLE
[sysrst_ctrl] Cleanup

### PR DESCRIPTION
**Pull request by marnovandermaas**
Remove obsolete TODO in DV and move theory of operation documentation to its own page.

### CODE REVIEW

**Review by crteja — APPROVED** [review]
Looks good to me

File: hw/ip/sysrst_ctrl/doc/theory_of_operation.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+Upon reset, the `flash_wp_l_o` signal will be asserted active low.
+The software can release `flash_wp_l_o` explicitly by setting [`PIN_OUT_CTL.FLASH_WP_L`](../data/sysrst_ctrl.hjson#pin_out_ctl) to 0 when needed.
+The `flash_wp_l_o` signal does have a corresponding input signal `flash_wp_l_i` - but that one is mainly intended for pad observability and does not have a bypass path to `flash_wp_l_o`.
+Hence, the value of `flash_wp_l_o` defaults to logic 0 when it is not explicitly driven via the override function.
+
+Note that since the `sysrst_ctrl` does not have control over the pad open-drain settings, software should properly initialize the pad attributes of the corresponding pad in the [pinmux configuration](../../pinmux/README.md) before releasing `flash_wp_l_o`.
+
+## Device Interface Functions (DIFs)
+
+- [Device Interface Functions](../../../sw/device/lib/dif/dif_sysrst_ctrl.h)
```
Comment (HU90m) [inline hw/ip/sysrst_ctrl/doc/theory_of_operation.md:170]:
```suggestion
- [Device Interface Functions](../../../../sw/device/lib/dif/dif_sysrst_ctrl.h)
```

File: hw/ip/sysrst_ctrl/doc/theory_of_operation.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+## Registers
+
+* [Register Table](data/sysrst_ctrl.hjson#registers)
```
Comment (HU90m) [inline hw/ip/sysrst_ctrl/doc/theory_of_operation.md:174]:
```suggestion
* [Register Table](../data/sysrst_ctrl.hjson#registers)
```

**Review by HU90m — CHANGES_REQUESTED** [review]
Found two broken links. Other than that, it looks good.

File: hw/ip/sysrst_ctrl/doc/theory_of_operation.md (hunk 1 above)
Comment (marnovandermaas) [reply hw/ip/sysrst_ctrl/doc/theory_of_operation.md]: Thanks, I've fixed that.

File: hw/ip/sysrst_ctrl/doc/theory_of_operation.md (hunk 2 above)
Comment (marnovandermaas) [reply hw/ip/sysrst_ctrl/doc/theory_of_operation.md]: Thanks, I've fixed that :)

**Review by HU90m — APPROVED** [review]

File: hw/ip/sysrst_ctrl/doc/theory_of_operation.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
+# Theory of Operations
```
Comment (andreaskurth) [inline hw/ip/sysrst_ctrl/doc/theory_of_operation.md:1]:
```suggestion
# Theory of Operation
```

According to the new file name and to be aligned with other modules

**Review by andreaskurth — APPROVED** [review]
LGTM overall, thanks for this cleanup and fixes @marnovandermaas!

I've gone through the changes in the moved *Theory of Operations* section. The link fixes not only include relative path changes (`../` added) but also `spi_host` -> `sysrst_ctrl` fixes, which is good. (The GitHub UI is rather useless to review that diff, so I used `meld` locally.)

I have one minor change request, otherwise I think this is ready

**Review by crteja — APPROVED** [review]
LGTM

File: hw/ip/sysrst_ctrl/doc/theory_of_operation.md (hunk 3 above)
Comment (marnovandermaas) [reply hw/ip/sysrst_ctrl/doc/theory_of_operation.md]: Thanks, this should be updated now.

**Pull request merged**